codes/quantum/qubits/CSS.yml

```diff
@@ -27,6 +27,7 @@ protection: 'Detects errors on \(d-1\) qubits, corrects errors on \(\left\lfloor
 
 notes:
   - 'Original requirement of \(C_X^\perp \subset C_Z\) \cite{arXiv:quant-ph/9512032} has been relaxed to absorb \hyperref[code:hypergraph_product]{hypergraph product} codes.'
+  - 'There exists a many to one mapping from size three chain complexes to CSS codes. A chain complex of size three is given by binary vector spaces \(C_2\), \(C_1\), \(C_0\) and boundry maps \(\partial_2\), \(\partial_1\) that satisfy \(\partial_1 \circ \partial_2 = 0\). We can construct a CSS code by attaching a physical qubit to every basis element of \(C_1\) and letting the stabilizers \(C_x\) and \(C_z\) be \(Im(\partial_1^T)\) and \(Im(\partial_2)\), respectivley. Then the normalizer of the stabilizers \(N(C_x)\) and \(N(C_z)\) are \(Ker(\partial_2^T)\) and \(Ker(\partial_1)\), repectivley. The number of encoded logical qubits is equal to the dimension of the first \(Z_2\)-Homology of the chain complex, \(H_1(\partial, Z_2) = \frac{Ker(\partial_1)}{Im(\partial_2)}\). Usually the chain complex used in this construction comes from the chain complex associated with a cellulation of a manifold. When the manifold is a two dimensional surface we use its entire chain. For larger dimensions we use the three largest non-trivial vector spaces in its chain. The distance of the CSS code is equal to the minimum of the combinatorial (\(d-1\))-systole of the cellulated \(d\)-dimensional manifold and its dual. For surfaces there is an upper bound of \(d \leq O(\sqrt(n))\) on the distance, therefore codes coming from higher dimensional spaces are studied for larger distance. Codes constructed in this manner are called Homological codes and capture all CSS codes. For more on Homological codes. \cite{arXiv:1802.01520} '
```

Comment:
What do you mean "attaching a physical qubit". You mean associating?

Reply:
Mention that boundary maps are just operators (matrices) on the binary vector space.

Reply:
Is there always a chain complex associated wiht a CSS code? What about CSS codes with unequal numbers of X and Z stabilizers? https://quantum-journal.org/papers/q-2019-02-04-120/pdf/ mentions only the equal case. This is important... if it's really not a one-to-one correspondence, then homological codes should be a child of CSS.

Reply:
This is true for all CSS codes I believe. Page 13 of https://arxiv.org/pdf/1311.0885.pdf talks about this.

Reply:
Thanks, I see it. Please mention the mapping in both directions explicitly, commenting on the many-to-one-ness. You can, e.g., discuss the CSS-to-chain mapping that in a new paragraph below the above discussion.

Reply:
Add Bombin ref in first sentence. Apparently that started it all.

Reply:
Systole sounds scary; Bravyi/Hastings 2013 say "cycle" and "cocycle". Why not use that?

Reply:
Just to clarify: "homological quantum codes" and "CSS codes"  are mathematically identical objects. There is no need to distinguish between them (see e.g. Section II.B in [https://arxiv.org/abs/2012.09271](url)). Further, homological products [https://arxiv.org/abs/1311.0885](url) / hypergraph products and tensor products [https://arxiv.org/abs/1512.07081](url) are pretty much identical operations. It might make sense to merge them into one article.

Regarding the word 'systole': it refers to closed loops of minimum length, whereas 'cycles' can be though of as collections of closed loops, so these terms are not interchangeable.